src/pg_cron.c

```diff
@@ -1263,7 +1274,7 @@ ManageCronTask(CronTask *task, TimestampTz currentTime)
 		case CRON_TASK_WAITING:
 		{
 			/* check if job has been removed */
-			if (!task->isActive)
+			if (!task->isActive || !LaunchActiveJobs)
```

Comment:
I'd expect an additional check to not load new tasks. Otherwise, it just keeps itself busy with adding and removing tasks from the hash for no reason.

Reply:
> I'd expect an additional check to not load new tasks

Good point, thanks! Done in the new revision.